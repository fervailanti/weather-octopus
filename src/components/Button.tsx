import React, { FC, ReactElement } from 'react'

// Button component is not being used right now, but will we in the future.

interface ButtonProps {
  action?: () => void
}

const Button: FC<ButtonProps> = ({ children, action }): ReactElement => {
  const transition = 'transition duration-150 ease-in-out'
  const hover = 'transform lg:hover:scale-105 lg:hover:bg-orange-900'
  const focus = 'focus:outline-none'
  const active = 'active:opacity-50'
  return (
    <button
      className={`bg-red-900 text-white rounded-lg shadow-lg py-2 px-4 m-4 ${hover} ${transition} ${focus} ${active}`}
      onClick={action}
    >
      {children}
    </button>
  )
}

export default Button
