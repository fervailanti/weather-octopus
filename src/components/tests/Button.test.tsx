import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Button from '../Button'

describe('Button Component', () => {
  it('Should render correctly', () => {
    const button = render(<Button>Test Button</Button>)
    expect(button.container).toMatchSnapshot()
  })
  it('Should execute action passed by props', () => {
    const dummyAction = jest.fn()
    render(<Button action={dummyAction}>Test Button</Button>)
    fireEvent.click(screen.getByText('Test Button'))
    expect(dummyAction).toHaveBeenCalledTimes(1)
  })
})
