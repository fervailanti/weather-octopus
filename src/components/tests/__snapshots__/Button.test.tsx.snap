// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Should render correctly 1`] = `
<div>
  <button
    class="bg-red-900 text-white rounded-lg shadow-lg py-2 px-4 m-4 transform lg:hover:scale-105 lg:hover:bg-orange-900 transition duration-150 ease-in-out focus:outline-none active:opacity-50"
  >
    Test Button
  </button>
</div>
`;
